pkg/controller/v1beta1/inferenceservice/components/predictor.go

```diff
@@ -189,9 +192,48 @@ func (p *Predictor) Reconcile(ctx context.Context, isvc *v1beta1.InferenceServic
 	} else {
 		var err error
 		podLabelKey = constants.RevisionLabel
+
 		if kstatus, err = p.reconcileKnativeDeployment(ctx, isvc, &objectMeta, &podSpec); err != nil {
 			return ctrl.Result{}, err
 		}
+		if p.forceStopRuntime {
+			// Exit early if we have already set the status to stopped
+			existing_stopped_condition := isvc.Status.GetCondition(v1beta1.Stopped)
+			if existing_stopped_condition != nil && existing_stopped_condition.Status == corev1.ConditionTrue {
+				return ctrl.Result{}, nil
+			}
+
+			// Clear all statuses
+			isvc.Status = v1beta1.InferenceServiceStatus{}
+
+			// Set the ready condition to false via the ready condition's dependencies PredictorReady and IngressReady
+			predictor_ready_condition := &apis.Condition{
+				Type:   v1beta1.PredictorReady,
+				Status: corev1.ConditionFalse,
+			}
+			isvc.Status.SetCondition(v1beta1.PredictorReady, predictor_ready_condition)
+
+			ingress_ready_condition := &apis.Condition{
+				Type:   v1beta1.IngressReady,
+				Status: corev1.ConditionFalse,
+			}
+			isvc.Status.SetCondition(v1beta1.IngressReady, ingress_ready_condition)
```

Comment:
I think this condition should be handled in `ingress_reconciler`.

Reply:
not against it per se, but is it cleaner to have it in the same place as the "Predictor ready" status since the end goal of setting both is to get the "Ready" status to report as false as these are its dependent statuses?

Reply:
I see this code is no longer present in the current revision. So, I assume you moved it around?

Reply:
yes it got moved to the ingress reconciler